src/Features/Core/Portable/CodeRefactorings/InvertIf/AbstractInvertIfCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,392 @@
+        {
+            if (operation is IConversionOperation conversion && conversion.IsImplicit)
+            {
+                return RemoveImplicitConversion(conversion.Operand);
+            }
+
+            return operation;
+        }
+
+        private bool CanSimplifyToLengthEqualsZeroExpression(
+            IOperation variableExpression,
+            ILiteralOperation numericLiteralExpression,
+            CancellationToken cancellationToken)
+        {
+            var numericValue = numericLiteralExpression.ConstantValue;
+            if (numericValue.HasValue && numericValue.Value is 0)
+            {
+                if (variableExpression is IPropertyReferenceOperation propertyOperation)
+                {
+                    var property = propertyOperation.Property;
+                    if ((property.Name == nameof(Array.Length) || property.Name == LongLength))
+                    {
+                        var containingType = property.ContainingType;
+                        if (containingType != null &&
+                            (containingType.SpecialType == SpecialType.System_Array ||
+                            containingType.SpecialType == SpecialType.System_String))
+                        {
+                            return true;
+                        }
+                    }
+                }
+
+                var type = variableExpression.Type;
+                if (type != null)
+                {
+                    switch (type.SpecialType)
+                    {
+                        case SpecialType.System_Byte:
+                        case SpecialType.System_UInt16:
+                        case SpecialType.System_UInt32:
+                        case SpecialType.System_UInt64:
+                            return true;
+                    }
+                }
+            }
+
+            return false;
+        }
+
+private SyntaxNode GetNegationOfLiteralExpression(
+    SyntaxNode expression,
+    SyntaxGenerator generator,
+    SemanticModel semanticModel)
+        {
+            var operation = semanticModel.GetOperation(expression);
+            SyntaxNode newLiteralExpression;
+            var leadingExpressionTrivia = expression.GetLeadingTrivia();
+            var trailingExpressionTrivia = expression.GetTrailingTrivia();
+
+            if (operation?.Kind == OperationKind.Literal && operation.ConstantValue.HasValue && operation.ConstantValue.Value is true)
+            {
+                newLiteralExpression = generator.FalseLiteralExpression();
+            }
+            else if (operation?.Kind == OperationKind.Literal && operation.ConstantValue.HasValue && operation.ConstantValue.Value is false)
+            {
+                newLiteralExpression = generator.TrueLiteralExpression();
+            }
+            else
+            {
+                newLiteralExpression = generator.LogicalNotExpression(expression);
```

Comment:
this seems wrong.  if 'expression' has trivia on it, you'll wrap that trivia, then add it on again afterwards because of the code below.